File: LeetCode/LeetCode 75/1480_Running_Sum_of_1d_Array/1480_Running_Sum_of_1d_Array.kt
class Solution {
    fun runningSum(nums: IntArray): IntArray {
        for (i in 1 until nums.size) {
            nums[i] += nums[i-1]
        }
        return nums
    }
}

fun main() {
    val s = Solution()
    val input = intArrayOf(1,2,3,4)
    print(s.runningSum(input))
}

File: LeetCode/LeetCode 75/1480_Running_Sum_of_1d_Array/1480_Running_Sum_of_1d_Array.py
from functools import reduce
from typing import *

class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        result = []
        sum = 0
        for num in nums:
            sum += num
            result.append(sum)
        return result


s = Solution()

nums = [1,2,3,4]
print(s.runningSum(nums))
